pkg/objstore/s3/s3.go

```diff
@@ -86,6 +89,7 @@ func NewBucket(conf *Config, reg prometheus.Registerer) (*Bucket, error) {
 	}
 
 	client, err := f(conf.Endpoint, conf.AccessKey, conf.SecretKey, !conf.Insecure)
+	client.SetAppInfo("Thanos", fmt.Sprintf("%s (%s)", version.Version, runtime.Version()))
```

Comment:
I think on the issue here https://github.com/improbable-eng/thanos/pull/294 we wanted extra info what component is this.  Maybe worth to add another parameter to `NewBucket` with additional `component string` and put it here? Not sure - just an idea.

Reply:
Should it be more like `Thanos-<component>` or `Thanos <component>`? I'd prefer the dashed version (I have a WIP jaeger tracing implementation which uses `thanos-component`). And also should the name be capitalized?

Reply:
I don't mind really any of these (: What do you think will be more readable for user? (I guess while debugging AWS access logs or something like that?)

I would avoid capitalizing it too much, but again - up to you. @mattbostock any preferences?

Reply:
For me dashes are easier, there are always issues with spaces/tabs (eg thanos-store, thanos-sidecar is easier to find than thanos<blank><maybesomething><blank><but not that>)